pkg/geoipupdate/config.go

```diff
@@ -120,10 +118,67 @@ func NewConfig( //nolint: gocyclo // long but breaking it up may be worse
 		Parallelism:       1,
 	}
 
+	// Override config with values from the config file.
+	err := setConfigFromFile(config, path)
+	if err != nil {
+		return nil, err
+	}
+
+	// Override config with values from environment variables.
+	err = setConfigFromEnv(config)
+	if err != nil {
+		return nil, err
+	}
+
+	// Override config with values from option flags.
+	err = setConfigFromFlags(config, flagOptions...)
+	if err != nil {
+		return nil, err
+	}
+
+	// Set config values that depend on other config values. For instance
+	// proxyURL may have been set by the default config, and proxyUserInfo
+	// by config file. Both of these values need to be combined to create
+	// the public Proxy field that is a *url.URL.
+
+	config.Proxy, err = parseProxy(config.proxyURL, config.proxyUserInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	if config.LockFile == "" {
+		config.LockFile = filepath.Join(config.DatabaseDirectory, ".geoipupdate.lock")
+	}
+
+	// Validate config values now that all config sources have been considered and
+	// any value that may need to be created from other values has been set.
+
+	err = validateConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	// Reset values that were only needed to communicate information between
+	// config overrides.
+
+	config.proxyURL = ""
```

Comment:
I'm not sure if we want to do this or not. It does make testing a little easier at least.